src/skytable.ts

```diff
@@ -0,0 +1,51 @@
+import { Socket } from 'node:net';
+import { TLSSocket } from 'node:tls';
+import { connectionWrite } from './connection';
+import { encodeParams, formatResponse } from './protocol';
+
+export type ConnectionOptions = {
+  port: number;
+  hostname: string;
+};
+
+export type ConnectionTlsOptions = ConnectionOptions & { certFile: string };
+
+export type ColumnBase = string | number | boolean | null | bigint;
+
+export type SQParam<T = ColumnBase> = T | SQParam<T>[];
+
+export type ColumnBinary = typeof Buffer;
+
+export type ColumnList<T> = T | ColumnList<T>[];
+
+export type Column =
+  | Buffer
+  | ColumnBase
+  | ColumnBinary
+  | ColumnList<ColumnBase>;
+
+export type Row = Column[];
+
+export type Rows = Row[];
+
+export type QueryResult = Column | Row | Rows;
+
+export function createSkytable(connection: Socket | TLSSocket) {
```

Comment:
If this function is exported, I think we'll want to call it something like `createConnection` or `connectDB`